File: PM Capstone 2/client/pmcapstone/src/properties/Property.js
import React from "react"
import { Link } from "react-router-dom"

export const Property = ({ property }) => {
    const isVacant = () => {
        if (property.vacant === false){
            return <td>N</td>
        }
        else return <td>Y</td>
    }
    return (
        <tbody>
            <tr>
                <td><Link to={`/properties/${property.id}`}> {property.streetAddress} </Link></td>
                <td> {property.city}, {property.state} </td>
                <td> {property.type} </td>
                <td> {property.sizeDescription}</td>
                <td> ${property.rent} </td>
                {isVacant()}
            </tr>
        </tbody>
    )
}

File: PM Capstone 2/Models/Property.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using PropertyManager.Models;

namespace PropertyManager.Models
{
    public class Property
    {
        public int Id { get; set; }

        [Required]
        public string StreetAddress { get; set; }

        [Required]
        public string City { get; set; }

        [Required]
        public string State { get; set; }

        [Required]
        public string Type { get; set; }

        public string SizeDescription { get; set; }

        public int Rent { get; set; }

        [Required]
        public bool Vacant { get; set; }

        public int? UserProfileId { get; set; }

        public UserProfile? UserProfile { get; set; }
    }
}



